clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/securitygroup/view/AzureSecurityGroupProvider.groovy

```diff
@@ -0,0 +1,118 @@
+import com.netflix.spinnaker.clouddriver.azure.resources.securitygroup.model.AzureSecurityGroup
+import com.netflix.spinnaker.clouddriver.azure.resources.securitygroup.model.AzureSecurityGroupDescription
+import com.netflix.spinnaker.clouddriver.model.SecurityGroupProvider
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class AzureSecurityGroupProvider implements SecurityGroupProvider<AzureSecurityGroup> {
+
+  private final AzureCloudProvider azureCloudProvider
+  private final Cache cacheView
+  final ObjectMapper objectMapper
+
+  @Autowired
+  AzureSecurityGroupProvider(AzureCloudProvider azureCloudProvider, Cache cacheView, ObjectMapper objectMapper) {
+    this.azureCloudProvider = azureCloudProvider
+    this.cacheView = cacheView
+    this.objectMapper = objectMapper
+  }
+
+  String getCloudProvider() {
+    return azureCloudProvider.id
+  }
+
+  @Override
+  String getType() {
+    return azureCloudProvider.id
+  }
+
+  @Override
+  Set<AzureSecurityGroup> getAll(boolean includeRules) {
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(azureCloudProvider, '*', '*', '*', '*'), includeRules)
+  }
+
+  @Override
+  Set<AzureSecurityGroup> getAllByRegion(boolean includeRules, String region) {
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(azureCloudProvider, '*', '*', region, '*'), includeRules)
+  }
+
+  @Override
+  Set<AzureSecurityGroup> getAllByAccount(boolean includeRules, String account) {
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(azureCloudProvider, '*', '*', '*', account), includeRules)
+  }
+
+  @Override
+  Set<AzureSecurityGroup> getAllByAccountAndName(boolean includeRules, String account, String name) {
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(azureCloudProvider, name, '*', '*', account), includeRules)
+  }
+
+  @Override
+  Set<AzureSecurityGroup> getAllByAccountAndRegion(boolean includeRules, String account, String region) {
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(azureCloudProvider, '*', '*', region, account), includeRules)
+  }
+
+  @Override
+  AzureSecurityGroup get(String account, String region, String name, String vnet) {
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(azureCloudProvider, name, '*', region, account), true)[0]
+  }
+
+  Set<AzureSecurityGroup> getAllMatchingKeyPattern(String pattern, boolean includeRules) {
+    loadResults(includeRules, cacheView.filterIdentifiers(Keys.Namespace.SECURITY_GROUPS.ns, pattern))
+  }
+
+  Set<AzureSecurityGroup> loadResults(boolean includeRules, Collection<String> identifiers) {
+    def transform = this.&fromCacheData.curry(includeRules)
+    def data = cacheView.getAll(Keys.Namespace.SECURITY_GROUPS.ns, identifiers, RelationshipCacheFilter.none())
+    def transformed = data.collect(transform)
+
+    return transformed
+  }
+
+  AzureSecurityGroup fromCacheData(boolean includeRules, CacheData cacheData) {
```

Comment:
AFAIK the deck calls into gate/orca/clouddriver will get all the data they need, including the rules (that is security rules). Once the remaining of the deck work is completed (edit/clone/delete) we should clean this up if it's not needed.